util/progress/progressui/printer.go

```diff
@@ -28,13 +30,71 @@ type lastStatus struct {
 
 type textMux struct {
 	w         io.Writer
+	printer   vertexPrinter
 	current   digest.Digest
 	last      map[string]lastStatus
 	notFirst  bool
 	nextIndex int
 	desc      string
 }
 
+type vertexPrinter interface {
+	printVtx(p *textMux, t *trace, v *vertex)
+}
+
+type vertexUpdate struct {
```

Comment:
As these are the types returned to user that supposedly someone will parse make these types public (can also put definition in another package for easier import).

I don't think this is `vertexUpdate`. It is more like a invocation.

Reply:
I've adapted the various `VertexX` types into their own structs that match this package's API a bit more cleanly. They're mostly based on the ones in the `client` package, but with some small differences and the correct JSON annotations.

Even when they're alike, it's probably better to keep them separate just so changing one of these types doesn't inadvertently affect the output of the other.

I think the `progressui` package shouldn't be a problem for most people to import, but we can move the type with a type alias if it ends up being a problem in the future or consumers can just copy the file with the JSON definitions pretty easily into their own projects.